src/index.js

```diff
@@ -0,0 +1,204 @@
+rome.setAirportMaxCapacity(-2); // not valid
+
+// Display airport
+displayAirport(berlin);
+displayAirport(rome);
+console.log('------------------------');
+
+// Ask if airport is full
+console.log('berlin full? ' + berlin.isAirportFull())
+console.log('rome full? ' + rome.isAirportFull())
+console.log('------------------------');
+
+//Land more planes
+berlin.instructToLand(realPlane1); // impossible - is already landed
+berlin.instructToLand(realPlane5);
+rome.instructToLand(realPlane6); // impossible - capacity is 2 
+
+// display airports
+displayAirport(berlin);
+displayAirport(rome);
+console.log('------------------------');
+
+// Plane at airport checks
+console.log(`Is plane A1 at Berlin? ` + berlin.checkPlaneAtAirport(realPlane1)); // true
+console.log(`Is plane F6 at Berlin? ` + berlin.checkPlaneAtAirport(realPlane6)); // false
+console.log(`Is plane C3 at Rome? ` + rome.checkPlaneAtAirport(realPlane3)); // true
+console.log('------------------------');
+
+rome.setWeatherSafeForAction(false); // Makes it impossible to takeoff from rome (Rome) as weather is stormy / unsafe
+
+//Take off planes
+berlin.instructToTakeOff2(realPlane1);
+berlin.instructToTakeOff2(realPlane3); // impossible at other airport/ not at Berlin
+rome.instructToTakeOff2(realPlane3); // impossible bad weather / unsafe
+
+// display airports
+displayAirport(berlin);
+displayAirport(rome);
+console.log('------------------------');
+
+// Show plane in flight cant take off
+console.log('Logic Demo 1 ')
+console.log('------------------------');
+
+logicPlane = new Plane('Grounded', true);
+logicAirport = new Airport('Makes No Sense');
+logicAirport.instructToLand(logicPlane);
+
+displayPlane(logicPlane);
+logicArray = logicAirport.getListOfPlanesAtAirport;
+console.log(JSON.stringify(logicArray));
+console.log('------------------------');
+
+// Change inflight to be true then ask to take off - then review same info as previous console.log
+logicPlane.setInFlight(true);
+logicAirport.instructToTakeOff2(logicPlane);
+
+displayPlane(logicPlane);
+logicArray = logicAirport.getListOfPlanesAtAirport;
+console.log(JSON.stringify(logicArray));
+console.log('------------------------');
+
+// Show plane grounded cant be ask to land
+console.log('Logic Demo 2 ')
+console.log('------------------------');
+
+logicPlane2 = new Plane('Grounded', false); // In flight is false so cant be asked to land
+logicAirport2 = new Airport('Makes No Sense');
+logicAirport2.instructToLand(logicPlane2);
+
+displayPlane(logicPlane2);
+logicArray2 = logicAirport2.getListOfPlanesAtAirport;
+console.log(JSON.stringify(logicArray2));
+
+console.log('------------------------');
+
+console.log('Weather generator Demo')
+console.log('------------------------');
+
+// True represents safe action weather i.e sunny, false represents stormy
+for (loop = 0; loop < 10; loop++) {
+    console.log(`Weather safe for landing/takeOff? ` + Airport.weatherGenerator());
+}
+
+function isThisAnInstanceOfPlane(input) {
```

Comment:
Maybe should have done a printer class instead of mixing Function based programming with Object Oriented